packages/flutter/lib/src/material/scaffold.dart

```diff
@@ -36,19 +37,94 @@ enum _ScaffoldSlot {
   statusBar,
 }
 
+/// A position that the [FloatingActionButton] can dock to in a [Scaffold].
+/// 
+/// [getOffset] 
+class FabPosition {
+  /// End-aligned FAB, floating at the bottom of the screen.
+  /// 
+  /// This is the default alignment of FABs in Material apps.
+  static const FabPosition endFloat = const _EndFloatFab();
+
+  /// Centered FAB, floating at the bottom of the screen.
+  static const FabPosition centerFloat = const _CenterFloatFab();
+
+  const FabPosition(this.getOffset);
```

Comment:
nit: in flutter, we always put the constructors first, that way we can tell at a glance if there's an implicit constructor. otherwise you'd have to scan the entire class to determine if it had an implicit constructor.

also, all public members should have documentation; see https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#avoid-useless-documentation for some writing prompts for writing docs when it's not clear what to write.

Reply:
Done.  Does this apply to constructors when there's only one unnamed constructor?

Reply:
Yup.